From 1419bbb0325b7ab83d60b02110bb256a4f5f0009 Mon Sep 17 00:00:00 2001
From: Bule <1052239551@qq.com>
Date: Fri, 2 Sep 2016 03:07:00 +0800
Subject: [PATCH] add mtk-ueventd

---
 core/rootdir/ueventd.rc | 140 ++++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 136 insertions(+), 4 deletions(-)

diff --git a/rootdir/ueventd.rc b/rootdir/ueventd.rc
index 4023ea6..d29720d 100644
--- a/rootdir/ueventd.rc
+++ b/rootdir/ueventd.rc
@@ -13,23 +13,93 @@ subsystem adf
 /dev/ashmem               0666   root       root
 /dev/binder               0666   root       root
 
+   #change partition permission
+/dev/preloader            0640   root       system
+/dev/bootimg              0640   root       system
+/dev/recovery             0660   root       system
+/dev/sec_ro               0640   root       system
+/dev/kb                   0660   root       system
+/dev/dkb                  0660   root       system
+/dev/seccfg               0660   root       system
+/dev/pro_info             0660	 root	    system
+/dev/otp 		  0660	 root	    system
+/dev/nvram 		   0660	 root	    system
+/dev/misc		   0660	 root	    system
+/dev/misc2		   0660	 root	    system
+/dev/block/platform/mtk-msdc.0/by-name/misc2		   0660	 root	    system
+/dev/block/mmcblk0	   0660	 root	    system
+/dev/block/mmcblk0boot0	   0660	 root	    system
+/dev/block/mmcblk0boot1	   0660	 root	    system
+/dev/misc-sd              0660   root       system
+
+
+#connectivity
+# Create char device file for WMT, GPS, BT, FM, WIFI
+#    mknod /dev/stpwmt c 190 0;
+#    mknod /dev/stpgps c 191 0;
+#    mknod /dev/stpbt  c 192 0;
+
+#    chmod 0660 /dev/stpwmt
+#    chown system system /dev/stpwmt
+#    chmod 0660 /dev/wmtdetect
+#    chown system system /dev/wmtdetect
+#
+# BT
+#    chmod 0660 /dev/stpbt
+#    chown bluetooth radio /dev/stpbt
+#
+# GPS
+#    chown gps gps /dev/stpgps
+#    chmod 0660 /dev/stpgps
+
+/dev/stpwmt              0660  system      system
+/dev/wmtdetect           0660  system      system
+/dev/stpbt               0660  bluetooth   radio
+/dev/stpgps              0660  gps         gps
+/dev/stpant				 0660  system      system
+
+# WIFI
+#  chown system system /dev/wmtWifi
+#  chmod  0660  /dev/wmtWifi
+
+/dev/wmtWifi             0660  system      system
+
+# FMRadio
+/dev/fm			 0660  system	   media
+
+# add the read write right of ttyUSB*
+/dev/bus/usb/*     	  0660   root       usb
+/dev/ttyUSB0              0660   radio      radio
+/dev/ttyUSB1              0660   radio      radio
+/dev/ttyUSB2              0660   radio      radio
+/dev/ttyUSB3              0660   radio      radio
+/dev/ttyUSB4              0660   radio      radio
+
 # Anyone can read the logs, but if they're not in the "logs"
 # group, then they'll only see log entries for their UID.
+/dev/xLog                 0600   root       root
 /dev/log/*                0666   root       log
+/dev/log/ksystem          0600   root       log
 
 # the msm hw3d client device node is world writable/readable
 /dev/msm_hw3dc            0666   root       root
 
-# the DIAG device node is not world writable/readable.
-/dev/diag                 0660   system     qcom_diag
-
 # gpu driver for adreno200 is globally accessible
 /dev/kgsl                 0666   root       root
 
+# Begin Immersion changes
+#/dev/tspdrv               0660   system       system
+# End Immersion changes
+
+#begin fpc fingerprint
+/dev/fpc1020              0666   system     system
+#end
+
 # kms driver for drm based gpu
 /dev/dri/*                0666   root       graphics
 
 # these should not be world writable
+/dev/diag                 0660   radio      radio
 /dev/diag_arm9            0660   radio      radio
 /dev/android_adb          0660   adb        adb
 /dev/android_adb_enable   0660   adb        adb
@@ -39,7 +109,7 @@ subsystem adf
 /dev/alarm                0664   system     radio
 /dev/rtc0                 0640   system     system
 /dev/tty0                 0660   root       system
-/dev/graphics/*           0660   system     graphics
+/dev/graphics/*           0660   root       graphics
 /dev/msm_hw3dm            0660   system     graphics
 /dev/input/*              0660   root       input
 /dev/eac                  0660   root       audio
@@ -85,6 +155,23 @@ subsystem adf
 /dev/usb_accessory        0660   root       usb
 /dev/tun                  0660   system     vpn
 
+/dev/ccci*                0660   radio       radio
+/dev/ttyC*                0660   radio       radio
+/sys/kernel/ccci          modem_info   0644   radio       radio
+/sys/kernel/ccci          md1_postfix  0644   radio       radio
+/sys/kernel/ccci          md2_postfix  0644   radio       radio
+/dev/eemcs*               0660   radio       radio
+/dev/emd*                 0660   radio       radio
+/dev/ccci_pcm_rx          0660   media      media
+/dev/ccci_pcm_tx          0660   media      media
+/dev/ccci_aud             0660   media      media
+/dev/ccci2_aud            0660   media      media
+/dev/eemcs_aud            0660   media      media
+
+# SGX device node
+/dev/pvrsrvkm             0666   root       root
+/dev/pvr_sync             0666   root       root
+
 # CDMA radio interface MUX
 /dev/ts0710mux*           0640   radio      radio
 /dev/ppp                  0660   radio      vpn
@@ -134,3 +221,48 @@ subsystem adf
 /sys/devices/system/cpu/cpufreq interactive/timer_slack      0664    system  system
 /sys/devices/system/cpu/cpufreq interactive/up_threshold_any_cpu_freq      0664    system  system
 /sys/devices/system/cpu/cpufreq interactive/up_threshold_any_cpu_load      0664    system  system
+/sys/devices/platform/leds-mt65xx/leds/green   delay_on      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/green   delay_off      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/red   delay_on      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/red   delay_off      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/blue   delay_on      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/blue   delay_off      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/yellow   delay_on      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/yellow   delay_off      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/white   delay_on      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/white   delay_off      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/cyan   delay_on      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/cyan   delay_off      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/violet   delay_on      0664  system   system
+/sys/devices/platform/leds-mt65xx/leds/violet   delay_off      0664  system   system
+
+#NFC device
+/dev/msr3110              0666   root       root
+/dev/mt6605               0660   nfc        radio
+
+#GPS
+/dev/gps	0660	gps	system
+
+#GPIO
+/dev/mtgpio	0600	radio	root
+
+#MTK BTIF driver
+/dev/btif 0600  system system
+
+#FOR VIA MODEM
+/dev/ttySDIO*	0660	radio	sdcard_rw
+/dev/ttyRB*     0660    radio   radio
+
+# Mali node
+/dev/mali                0666    system graphics
+
+#touch
+/dev/touch              0660   root  system
+/dev/hotknot            0660   root  system
+
+#hang_detect
+/dev/RT_Monitor         0600   system  system
+/dev/kick_powerkey      0600   system  system
+
+#MTK In-House TEE
+/dev/ttyACM0		  0660   radio	    radio
-- 
2.8.1

